rfcs/rfc-colin-states-mapping-to-lear.md

```diff
@@ -0,0 +1,163 @@
+- Start Date: 2021-12-03
+- Target Major Version: 
+- Reference Issues: [10098](https://github.com/bcgov/entity/issues/4797)
+- Entity Issue: [8285](https://github.com/bcgov/entity/issues/14691)
+- Implementation PR: (leave this empty)
+
+# Summary
+
+COLIN represents a business state as two states which are referred to as an operational state(`CORP_OP_STATE` table) and a corp state(`CORP_STATE`) table.  In many instances, these two states are used in conjunction to drive business logic and to determine allowable business state transitions between `ACTIVE` and `HISTORICAL` states for a given business.
+
+LEAR will not be adopting the COLIN approach of using the `CORP_STATE` table.  
+
+This RFC will provide design details as to how LEAR will achieve the same functionality as COLIN without the use of the `CORP_STATE` table.
+
+Note that the [Business States RFC](rfc-business-states.md) as well the current implementation of the Legal API was used as a starting point for the creation of this document.
+
+#### COLIN Database Structure for Business States
+![colin_states.png](rfc-colin-states-mapping-to-lear/colin_states.png)
+
+#### Current Corp State Data in CPRD
+_`ACT/HIS` column is `CORP_OP_STATE` table._
+_`Type` column is `CORP_STATE` table._
+![colin_corp_state_stats.png](rfc-colin-states-mapping-to-lear/colin_corp_state_stats.png)
+
+
+# Basic design
+
+The COLIN business states can be mapped 1-1 to the business states(`businesses.state`) in LEAR.  Only `ACTIVE` and `HISTORICAL` are valid business states.
+
+The `CORP_STATE` table in COLIN will not be used. It was essentially used track the latest filing a business had filed and not an actual state.  This functionality will be achieved in LEAR by using a combination of the filing type, dynamic calculations (warning structures or business properties like good standing) or using the filing type that transitioned a business into its current state.
+
+Compliance is informational only and does not block any registry actions but must be noted to viewers of the Registry.
+
+Good standing has various statuses and can lead to a business state transition.
```

Comment:
Statuses? Where will those be stored (so that they aren't confused with business state)? What will that look like?

Reply:
I'm not entirely sure.  I'm assuming that good standing might not be as simple as what we have where it's just true or false. Or it can be like that but we need to be able to provide more granular info to the API consumer as well as for the API logic.  

@thorwolpert This was just my interpretation of your comment about good standing having various statuses.  Please let me know if you were thinking something else.

Reply:
I'd love to see a table with all possible business "states" and each variable's value for that, eg:

| Business State | `state` | `isFrozen` | `isGoodStanding` |
| --- | --- | --- | --- |
| everything ok | ACTIVE | false | false |

etc

Reply:
Thanks for the examples!